tests/test_parameters.py

```diff
@@ -251,6 +251,25 @@ def test_daily_profile_leap_day(model):
     model.run()
     assert_allclose(inpt.flow, 365)
 
+def test_weekly_profile(simple_linear_model):
+    model = simple_linear_model
+
+    model.timestepper.start = "2004-01-01"
+    model.timestepper.end = "2005-05-01"
+    model.timestepper.delta = 7
+
+    values = np.arange(0, 52) ** 2 + 27.5
+
+    p = WeeklyProfileParameter.load(model, {"values": values})
+
+    @assert_rec(model, p)
+    def expected_func(timestep, scenario_index):
+        week = int(min((timestep.dayofyear - 1) // 7, 51))
```

Comment:
Perhaps make the test use the pandas implementation of `.week` to see if these are the same?